### PR TITLE
Add back mosml, most recent version with clean build.

### PR DESCRIPTION
**Pull request by FranklinChen**
Address https://github.com/Homebrew/homebrew/issues/17210 properly with a new formula.

### CODE REVIEW

File: Library/Formula/mosml.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+require "formula"
```
Comment (bfontaine) [inline Library/Formula/mosml.rb]: You can remove this line.

File: Library/Formula/mosml.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+                     "CC=#{ENV.cc}",
+                     "world"
+      system "make", "PREFIX=#{prefix}",
+                     "CC=#{ENV.cc}",
+                     "install"
+    end
+  end
+
+  test do
+    system "mosml", "-P full"
```
Comment (bfontaine) [inline Library/Formula/mosml.rb]: Please use `"#{bin}/mosml"`.

File: Library/Formula/mosml.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+class Mosml < Formula
+  homepage "http://mosml.org"
+  url "https://github.com/kfl/mosml/archive/ver-2.10.1.tar.gz"
+  sha1 "e53fa82074d1c60499dc4ca83521c229e0655ccc"
```
Comment (DomT4) [inline Library/Formula/mosml.rb]: This must use `sha256`.

File: Library/Formula/mosml.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+  depends_on "gmp"
+
+  def install
+    cd 'src' do
```
Comment (DomT4) [inline Library/Formula/mosml.rb]: Double quotes everywhere.

File: Library/Formula/mosml.rb [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+      system "make", "PREFIX=#{prefix}",
```
Comment (DomT4) [inline Library/Formula/mosml.rb]: Let's put this onto one line.

`system "make", "PREFIX=#{prefix}", "CC=#{ENV.cc}", "world"`

And such.

File: Library/Formula/mosml.rb [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+                     "CC=#{ENV.cc}",
+                     "world"
+      system "make", "PREFIX=#{prefix}",
```
Comment (DomT4) [inline Library/Formula/mosml.rb]: This can be one line too.

**Comment by DomT4** [comment]
```
==> brew style mosml

1 file inspected, no offenses detected
Error: 1 problem in 1 formula

==> audit problems
mosml:
 * Formula should have a desc (Description).
```

**Comment by FranklinChen** [comment]
Is this finally good to go? I'll audit any new formula contributions of mine carefully in the future to avoid the iterations that happened with this one!

**Comment by DomT4** [comment]
WFM. The previously troublesome header is no longer installed directly into `include` so :+1:.

Merged in https://github.com/Homebrew/homebrew/commit/803c3bc205853b086fb4799e6457206425cf97f7. Thank you for your contribution to Homebrew, we appreciate it! :smiley_cat:.

Just for future reference, ideal commit message here would have been `mosml 2.10.1 (new formula)`, Thanks!

**Pull request closed without merging**